xDrip/Components/Views/GlucoseCurrentInfoView.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  GlucoseCurrentInfoView.swift
+//  xDrip
+//
+//  Created by Dmitry on 6/16/20.
+//  Copyright © 2020 Faifly. All rights reserved.
+//
+
+import UIKit
+@IBDesignable
+class GlucoseCurrentInfoView: UIView, NibLoadable {
+    @IBOutlet private weak var glucoseIntValueLabel: UILabel!
+    @IBOutlet private weak var glucoseDecimalValueLablel: UILabel!
+    @IBOutlet private weak var slopeArrowLabel: UILabel!
+    @IBOutlet private weak var lastScanTitleLabel: UILabel!
+    @IBOutlet private weak var lastScanValueLabel: UILabel!
+    @IBOutlet private weak var difTitleLabel: UILabel!
+    @IBOutlet private weak var difValueLabel: UILabel!
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupFromNib()
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupFromNib()
+    }
+    
+    func setup(glucoseValue: Double?, slopeValue: Double?, lastScanDate: Date?, difValue: Double?) {
+        let components = getComponentsForGlucoseValue(glucoseValue ?? 0.0)
```

Comment:
If you allow nil values, you should handle them better, not just replacing with `0.0`. A good solution here would be displaying something like `--` instead of values.